pytorch_lightning/metrics/functional/classification.py

```diff
@@ -998,12 +1000,37 @@ def iou(
         tensor(0.4914)
 
     """
+    num_classes = get_num_classes(pred=pred, target=target, num_classes=num_classes)
+
     tps, fps, tns, fns, sups = stat_scores_multiple_classes(pred, target, num_classes)
-    if remove_bg:
-        tps = tps[1:]
-        fps = fps[1:]
-        fns = fns[1:]
-    denom = fps + fns + tps
-    denom[denom == 0] = torch.tensor(FLOAT16_EPSILON).type_as(denom)
-    iou = tps / denom
-    return reduce(iou, reduction=reduction)
+
+    scores = torch.zeros(num_classes, device=pred.device, dtype=torch.float32)
+
+    for class_idx in range(num_classes):
+        # Skip this class if its index is being ignored.
+        if class_idx == ignore_index:
+            continue
+
+        tp = tps[class_idx]
+        fp = fps[class_idx]
+        fn = fns[class_idx]
+        sup = sups[class_idx]
+
+        # If this class is not present in either the target (no support) or the pred (no true or false positives), then
+        # use the not_present_score for this class.
+        if sup + tp + fp == 0:
```

Comment:
in your comment you said it should be `(sup==0) or (tp+fp == 0)` but this line evaluates to `(sup == 0) and (tp+fp==0)

Reply:
Ah, I was using `not present in either A or B` to mean `and` not `or`, which I think is correct but is a bit unclear regardless. I updated the wording in https://github.com/PyTorchLightning/pytorch-lightning/pull/3098/commits/f206f23c2fa4752057e00d89f68f5d6aa6ff22d6 to be more clear, though less concise. Let me know if this could still be clarified.